src/components/AppProvider/AppProvider.scss

```diff
@@ -3,6 +3,7 @@
 // components then this css shall always be present.
 
 @import '../../styles/common';
+@import '../../styles/utilities.scss';
```

Comment:
Not sure if this is the right place to include the global utility classes?

Reply:
I think `global.scss` might be a good file name.

Reply:
Gotta admit, I'm not a fan of polaris exposing arbitrary global styles that people can leverage. The current global styles in this file has already felt a bit clunky; we don't use `:global` for classes anywhere else and I'm not convinced VisuallyHidden is a good enough reason to start. Global styles have always felt kinda weird in our react apps to me our components are set up to be self contained and you only use pull in the styles when you use the components. With global styles it isn't clear where the class names we depend upon come from - there's no proof that any global style we reference in a component is actually present, which feels fragile.


I think I'd prefer for this to be kept as a mixin, but perhaps we remove it from _public-api, and we keep it in _common - so that it is only usable internally, but not by external consumers. That way we're still shrinking our external surface area.


If you do decide to stick with the global class, yes in here is the best place. I think I'd prefer the class to be included directly in this file rather than imported from an external file.

Reply:
I think there is currently two use cases where global css could be considered. visually hidden and hide on print. I do want to reduce the need for rendering `<span>` elements to achieve this. I wonder if with strict guidelines and rules this would be considered approach.

Reply:
@BPScott good points. Something like this?

```
import styles from './Checkbox.scss';
import utilityStyles from '../../styles/utilities.scss';
```

```
    const inputClassName = classNames(
      styles.Input,
      utilityStyles.VisuallyHidden,
      isIndeterminate && styles['Input-indeterminate'],
      keyFocused && styles.keyFocused,
    );
```

The rendered html looks like this, which is still somewhat legible.

```
<input id="PolarisCheckbox2" type="checkbox" class="Checkbox-Input_1uEY6 utilities-VisuallyHidden_w6VoC" aria-invalid="false" role="checkbox" aria-checked="false" value="">
```
I do like the separate class because it surfaces right in the dom that there are a group of reusable properties being applied to this element, without having to go into the code to see that they are coming from a mixin.

If they get imported into components, then they don't need to be global anymore either. Seems like this approach addresses the concerns with global styles, while still allowing us to deprecate the mixin?